### ISSUE TITLE
2.0.2 copies partials when building.

### ISSUE THREAD

**Issue by magnus-bergman** [dlmanning/gulp-sass#306]
I just updated from 2.0.1 to 2.0.2, and now all my subfolders and partials are copied to the output folder when I build.

Here's the task I'm currently running which provides different results between 2.0.1 and 2.0.2:

```
var gulp = require('gulp');
var sass = require('gulp-sass');
var postcss = require('gulp-postcss');
var autoprefixer = require('autoprefixer-core');
var mqpacker = require('css-mqpacker');
var csswring = require('csswring');
var handleErrors = require('../util/handleErrors');

gulp.task('sass:production', function() {
  var processors = [
    autoprefixer({browsers: ['last 2 version']}),
    mqpacker,
    csswring
  ];

  return gulp.src('app/assets/styles/**/*.{sass,scss}')
    .pipe(sass({
      errLogToConsole: true,
      includePaths: ['bower_components']
    }))
    .on('error', handleErrors)
    .pipe(postcss(processors))
    .pipe(gulp.dest('build/assets/styles'));
});
```

Did something change or am I doing something wrong?

Thanks for all your hard work!

Edit: This is only have affecting empty files. I tried running it in a boilerplate of mine where I've scaffolded out a folder structure with some empty files. If I revert the changes from https://github.com/dlmanning/gulp-sass/commit/7f6605e433a32ecc01ed84b0024d7e6fcf7a2ed3 it works like before.

**Comment by atdrago** [comment]
I'm also having this exact problem, which I made reference to in https://github.com/dlmanning/gulp-sass/issues/305

The other odd thing: Partials and subfolders are copied to the output folder as `scss` files, not `css` files.

**Comment by edtorba** [comment]
+ 1

**Comment by Snugug** [comment]
This is… weird. We didn't change anything that should have affected this (in fact, I tried reverting back the change we made with partials and it's still happening). I'm digging in

**Comment by Snugug** [comment]
Are all of the partials that are being moved over empty? Or do they have content?

I've got a few tests running and it seems as if only empty partials are being moved over for me. I think it has to do with https://github.com/dlmanning/gulp-sass/blob/master/index.js#L28-L30, where we need to have it work the same way of our partial ignore, or move our partial detection code above that.

@dlmanning @Keats thoughts?

**Comment by Snugug** [comment]
This is also one of those nifty things we can't effectively automatically test for because we can't check in empty files :\

**Comment by edtorba** [comment]
Moving partial detection code https://github.com/dlmanning/gulp-sass/blob/master/index.js#L31-L33 above https://github.com/dlmanning/gulp-sass/blob/master/index.js#L28-L30, seemed to fix the issue.

**Comment by Snugug** [comment]
@edtorba Can you confirm that the only partials being moved over for you are empty ones?

**Comment by edtorba** [comment]
@Snugug Yes, only empty ones.

**Comment by Snugug** [comment]
Just released 2.0.3 that resolves this.
